### PR TITLE
refactor(opencrud-schema-provider): transform to typescript

### PR DESCRIPTION
**Pull request by SegevHaviv**

### CODE REVIEW

File: packages/opencrud-schema-provider/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import graphqlSchemaUtilsOriginal from './schemaUtils/graphqlSchemaUtils';
+import introspectionUtilsOriginal from './schemaUtils/introspectionUtils';
+
+export const openCrudDataModel = openCrudDataModelOriginal;
+export const openCrudSchema = openCrudSchemaOriginal;
+export const openCrudSchemaGraphql = openCrudSchemaGraphqlOriginal;
+export const getOpenCrudIntrospection = getOpenCrudIntrospectionOriginal;
+export const graphqlSchemaUtils = graphqlSchemaUtilsOriginal;
+export const introspectionUtils = introspectionUtilsOriginal;
+
```
Comment (shasaf1986) [inline packages/opencrud-schema-provider/src/index.ts:16]: we we also need to export default it ?

File: packages/opencrud-schema-provider/src/schemaUtils/graphqlSchemaUtils.ts [hunk 2]
```diff
@@ -42,16 +35,16 @@ function isList(fieldType) {
...
 
-function getFieldType(schema, entityName, fieldName) {
-  const relatedType = getField(schema, entityName, fieldName);
+export function getFieldType(schema, entityName, fieldName) {
+  const relatedType: any = getField(schema, entityName, fieldName);
   return relatedType.type.name || relatedType.type.ofType.name || relatedType.type.ofType.ofType.name;
 }
 
-module.exports = {
+export default {
```
Comment (shasaf1986) [inline packages/opencrud-schema-provider/src/schemaUtils/graphqlSchemaUtils.ts:47]: why export default ?

File: packages/opencrud-schema-provider/src/schemaUtils/introspectionUtils.ts [hunk 3]
```diff
@@ -111,23 +111,23 @@ function verifyExactlyOneFieldOfTypeOnEntity(childFieldsOfTypeCount, childType,
...
 
-function findTypeInIntrospection(introspection, resource) {
+export function findTypeInIntrospection(introspection, resource) {
   return introspection.types.find(t => {
     return t.name === resource;
   });
 }
 
-module.exports = {
+export default {
```
Comment (shasaf1986) [inline packages/opencrud-schema-provider/src/schemaUtils/introspectionUtils.ts:126]: why export default ?

File: packages/sequelize-model/src/index.ts [hunk 4]
```diff
@@ -1,14 +1,13 @@
 import sqOriginal from './sequelizeInit';
```
Comment (shasaf1986) [inline packages/sequelize-model/src/index.ts:1]: you just call it sq and re export it

File: packages/sequelize-model/src/index.ts [hunk 5]
```diff
@@ -1,14 +1,13 @@
...
-import { baseModel as OriginalBaseModel} from './base/baseModel';
+import { baseModel as OriginalBaseModel } from './base/baseModel';
 import { getCommonDbColumnDefinitions as originalGetCommonDbColumnDefinitions } from './commonColumnDefintions';
 
 export const baseModel = OriginalBaseModel;
 export const getCommonDbColumnDefinitions = originalGetCommonDbColumnDefinitions;
 export const sq = sqOriginal;
 
 export default {
```
Comment (shasaf1986) [inline packages/sequelize-model/src/index.ts:9]: export default ?

**Comment by SegevHaviv** [comment]
> I think that you do not need use export default at all ...

It's problematic. The usages ( in this repo and and on other consumers ) are either with const { x,y,z } = reuqire('...') or const X = require('...) this hybrid of named & default export was allowed cause it was basically all exported as default but then the { } const destruct worked for us and we got lucky on this one, that's not the case on esm imports. Therefore to keep backwards compatibility highest as possible, came up with this.

File: packages/sequelize-model/src/index.ts (hunk 4 above)
Comment (SegevHaviv) [reply packages/sequelize-model/src/index.ts:1]: I can but I prefer to keep the same semantic for all the imports / exports.

**Pull request merged**